packages/cli/src/util/link/setup-and-link.ts

```diff
@@ -51,23 +48,20 @@ export default async function setupAndLink(
     link,
     successEmoji = 'link',
     setupMsg = 'Set up',
-    projectName,
+    projectName = basename(path),
   }: SetupAndLinkOptions
 ): Promise<ProjectLinkResult> {
-  const { localConfig, config } = client;
+  const { config } = client;
 
-  const isFile = !isDirectory(path);
-  if (isFile) {
+  if (!isDirectory(path)) {
     output.error(`Expected directory but found file: ${path}`);
     return { status: 'error', exitCode: 1, reason: 'PATH_IS_FILE' };
   }
   if (!link) {
     link = await getLinkedProject(client, path);
   }
   const isTTY = client.stdin.isTTY;
-  const quiet = !isTTY;
   let rootDirectory: string | null = null;
-  let sourceFilesOutsideRootDirectory = true;
```

Comment:
Always default to `true` for new Projects at the API level. No need to duplicate that default on the client-side.